### PR TITLE
Added lists/arrays/hashes for go

### LINKED ISSUE #284: Add Go Lists/Arrays/Hashes
# Description

Lists/Arrays/Hashes for Go aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/lists.json` to the right Go directory in `web/thesauruses/`
* Edit the lists.json file to accurately reflect the latest major version of Go
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by Skr-suyash**
## What GitHub issue does this PR apply to?

Resolves #284

## What changed and why?

Added the Lists/Arrays/Hashes concept for Go language.

## Checklist

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by samuelburns** [comment]
Did this run properly when you ran it in the container? Mine didn't for swift

**Comment by Skr-suyash** [comment]
I tested it using django webserver and it was running fine.

Screenshot (36)

Let me test for the container too.

**Comment by geekygirlsarah** [comment]
> 
> 
> Did this run properly when you ran it in the container? Mine didn't for swift

This can be tested at https://codethesaurus-pr-373.herokuapp.com

File: web/thesauruses/go/lists.json [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+    "Functions On List Elements": [
+      "map",
+      "filter",
+      "reduce_left",
+      "reduce_right"
+    ]
+  },
+  "lists": {
+    "name_of_ordered_mutable_list": {
+      "name": "What is a ordered, mutable list called?",
```
Comment (geekygirlsarah) [inline web/thesauruses/go/lists.json]: You can delete all the `name` fields as they're pulled from the meta language file.

File: web/thesauruses/go/lists.json [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+      "name": "Swap two elements",
+      "code": "s[index1], s[index2] = s[index2], s[index1]"
+    },
+    "delete_ordered_mutable_list": {
+      "name": "Delete the list",
+      "code": "s = nil"
+    },
+    "name_of_unordered_mutable_list": {
+      "name": "What is a unordered, mutable list called?",
+      "code": "",
```
Comment (geekygirlsarah) [inline web/thesauruses/go/lists.json]: To help clean it up more, you can remove the `code` line when things aren't implemented in the language.

File: web/thesauruses/go/lists.json [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+      "name": "Concatenate two lists together",
+      "code": "final_list := append(list1, list2...)"
+    },
+    "split_list_at_index": {
+      "name": "Split lists at an index",
+      "code": "s1 := s[0:index+1] \ns2 := s[index+1:]"
+    },
+    "split_list_at_value": {
+      "name": "Split list at a value",
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/go/lists.json]: You might have missed this one. Is there code or is it `not-implemented`?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
A few quick, minor things and I think this could be good to merge.

File: web/thesauruses/go/lists.json (hunk 1 above)
Comment (Skr-suyash) [reply web/thesauruses/go/lists.json]: Done...

File: web/thesauruses/go/lists.json (hunk 3 above)
Comment (Skr-suyash) [reply web/thesauruses/go/lists.json]: Done...

**Comment by Skr-suyash** [comment]
Completed all the changes that were expected.

File: web/thesauruses/go/lists.json [hunk 4]
```diff
@@ -122,406 +122,269 @@
...
     "shuffle_list": {
-      "name": "Shuffle list elements",
       "comment": "Need to import standard library \"math/rand\"",
       "code": "rand.Shuffle(len(list), func(i, j int) { list[i], list[j] = list[j], list[i] })"
     },
     "reverse_list": {
-      "name": "Reverse order of list elements",
-      "code": "",
-      "not-implemented": true
+      "code": "for i := 0; i < len(numbers)/2; i++ {\n    j := len(numbers) - i - 1\n    numbers[i], numbers[j] = numbers[j], numbers[i]\n}"
```
Comment (geekygirlsarah) [inline web/thesauruses/go/lists.json]: I see what you're trying to do here, but if it's not a built-in function, it should just be "not implemented"

**Comment by Skr-suyash** [comment]
Removed the requested code. Go offers the bare minimum with its data structures, so I thought to include those. 😅

**Comment by geekygirlsarah** [comment]
Uh oh... I think you took out the algorithm AND the built-in libraries... 😬

I was trying to keep CT to the built-in stuff. Standard Libraries are built-in stuff. Third-party libraries are not, and writing algorithms are not.

Would it be terrible if you added those back in? You could probably cherry-pick from one of your old commits.

image

**Comment by Skr-suyash** [comment]
This means I should remove only that self implemented reverse algorithm and add the standard libraries back up.
Sorry for the trouble, I got a bit confused.

**Comment by Skr-suyash** [comment]
Kindly review the latest commit.

**Comment by geekygirlsarah** [comment]
I think it looks good @Skr-suyash ! I'll merge it in. Thank you for working on this!

**Pull request merged**